code/game/objects/items/food/egg.dm

```diff
@@ -85,6 +85,24 @@
 	else
 		..()
 
+/obj/item/food/egg/afterattack_secondary(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+
+	if(istype(target,/obj/machinery/griddle))
+
+		var/atom/broken_egg = new /obj/item/food/rawegg(target.loc)
+		broken_egg.pixel_x = src.pixel_x
+		broken_egg.pixel_y = src.pixel_y
+		playsound(get_turf(user), 'sound/items/sheath.ogg', 40, TRUE)
+		reagents.copy_to(broken_egg,reagents.total_volume)
+
+		var/obj/machinery/griddle/hit_griddle = target
+		hit_griddle.AddToGrill(broken_egg,user)
+		to_chat(user, "You crack [src] open onto [target].")
+
+		qdel(src)
+
+
```

Comment:
Additionally, you should return a `SECONDARY_ATTACK_X` call depending on success or not. 
If an egg was cracked, `return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN`
If no egg was cracked, return `SECONDARY_ATTACK_CALL_NORMAL`

Reply:
Ah, the attack chain things are all a little new - I'll get on it to ensure no strange behaviour can happen.